tests/function/test_config.py

```diff
@@ -190,3 +190,126 @@ def test_reporter_id_in_virtwho_conf(self, virtwho, globalconf, ssh_host):
                 and result['send'] == 1
                 and result['thread'] == 1
                 and result['reporter_id'] == reporter_id)
+
+    def test_log_per_config_in_virtwho_conf(self, virtwho, globalconf, hypervisor_handler, ssh_host):
+        """Test the log_per_config option in /etc/virtwho.conf
+
+        :title: virt-who: config: test log_per_config option
+        :id: 85accd49-54dc-4899-a9cf-c6fb07b2fe3c
+        :caseimportance: High
+        :tags: tier1
+        :customerscenario: false
+        :upstream: no
+        :steps:
+
+            1. Run virt-who with log-per-config=False in /etc/virt-who.conf
+            2. Run virt-who with log-per-config=True in /etc/virt-who.conf
+
+        :expectedresults:
+
+            1. unexpected /var/log/rhsm/virtwho* files not exist
+            2. Succeeded to find virtwho.destination_-*.log, virtwho.main.log, virtwho.main.log and
+            virtwho.rhsm_log.log file in /var/log/rhsm/
+        """
+        guest_uuid = hypervisor_handler.guest_uuid
+        globalconf.update('global', 'debug', 'True')
+
+        globalconf.update('global', 'log_per_config', 'False')
+        result = virtwho.run_service()
+        assert (result['error'] == 0
+                and result['send'] == 1
+                and result['thread'] == 1)
+        ret, output = ssh_host.runcmd('ls /var/log/rhsm/virtwho*')
+        assert ret is not 0
+
+        globalconf.update('global', 'log_per_config', 'True')
+        result = virtwho.run_service()
+        assert (result['error'] == 0
+                and result['send'] == 1
+                and result['thread'] == 1)
+        ret, files = ssh_host.runcmd('ls /var/log/rhsm/virtwho*')
+        assert (ret == 0
+                and 'virtwho.destination' in files
+                and 'virtwho.main.log' in files
+                and 'virtwho.rhsm_log.log' in files
+                and 'virtwho.virt.log' in files)
+
+        # assert the contents for the log files
+        for file in files.strip().split('\n'):
```

Comment:
I suggest to assert the contents in each log file in more detail.
Such as when check the virtwho.destination file, we should check below points,

- `guest uuid`, which is to make sure mapping is here
- `[virtwho.destination` keyword
- no other `[virtwho.rhsm_log`, `[virtwho.main` keywords

Reply:
Updated